### ISSUE TITLE
Usability of DataTable_<ETX, ETY>::getRow() and appendRow()

### ISSUE THREAD

**Issue by tkuchida** [opensim-org/opensim-core#1409]
Likely an issue that can be addressed by adding an example to the doxygen for `DataTable_`. Here's a MWE:
``` cpp
TimeSeriesTableVec3 tableA;
tableA.setColumnLabels({"col0", "col1"});
SimTK::RowVector_<Vec3> row{2, Vec3(0.1, 0.2, 0.3)};
tableA.appendRow(1.0, row);

TimeSeriesTableVec3 tableB;
tableB.setColumnLabels({"col0", "col1"});
tableB.appendRow(tableA.getIndependentColumn()[0],
    tableA.getRowAtIndex(0)); //looks like it should work, but doesn't
tableB.appendRow(tableA.getIndependentColumn()[0],
    static_cast<SimTK::RowVector_<Vec3>>(tableA.getRowAtIndex(0))); //works
```
This user interface is slightly awkward since it seems like `tableA.getRowAtIndex()` should return an object that is consumable by `tableB.appendRow()`. Similar to #1384.

**Comment by klshrinidhi** [comment]
All this is mostly to do with `DataTable` not hiding its implementation details and using SimTK objects in its interface. I think the interface of `DataTable` is not good. It never was.

`getRowAtIndex` and `getRow` return a `SimTK::RowVectorView`, which is returned by `SimTK::Matrix` for efficiency. `RowVectorView` is a proxy class and provides a *view* into a block of the underlying matrix.  Whereas `appendRow` expects a `SimTK::RowVector`. You should have seen a compiler error (which I think is the earliest I can catch this problem).

Regarding your use of `static_cast`, I am not sure if it the right thing to do because `RowVectorView` and `RowVector` are siblings in the hierarchy. `static_cast` will blow up at run-time if I correct.

**Solution:**
I can add an overload to `appendRow` to accept a `RowVectorView`. It should be pretty simple fix. Will get back with a PR.

**Comment by klshrinidhi** [comment]
Also worth mentioning is that my PR #1372 that I merged today adds functions `removeRowAtIndex` and `removeRow` if that can help avoid the above snippet until I add the overload.

**Comment by tkuchida** [comment]
> `static_cast` will blow up at run-time if I correct.

`static_cast` is the only solution I have found that works (in Release mode, at least :wink:).

> I can add an overload to `appendRow` to accept a `RowVectorView`. It should be pretty simple fix. Will get back with a PR.

Sounds good. Thanks!

**Comment by klshrinidhi** [comment]
> static_cast is the only solution I have found that works (in Release mode, at least ￼).

I think you are getting lucky. It will :dagger: you.

**Comment by tkuchida** [comment]
> I think you are getting lucky. It will :dagger: you.

Thanks. I'll be sure to update my code once #1410 has been merged. For now, a `SimTK_ASSERT_ALWAYS()` will let me know if the duct tape peels off :wink:

**Comment by aseth1** [comment]
closed by #1410
